src/librustc_mir/interpret/validation.rs

```diff
@@ -175,7 +175,7 @@ impl<'a, 'tcx, M: Machine<'tcx>> EvalContext<'a, 'tcx, M> {
                 }
             }
 
-            let result = self_.resolve_type_vars_if_possible(result);
+            let result = self_.resolve_type_and_region_vars_if_possible(result);
```

Comment:
You mean the one I removed in <https://github.com/solson/miri/pull/301/commits/c39bc68f6b0c714f7a9cccf76f405be3a0a5a1a2> because it broke some testcases?